lib/metasploit/framework/credential_collection.rb

```diff
@@ -257,29 +257,29 @@ def add_public(public_str='')
     # @yieldparam credential [Metasploit::Framework::Credential]
     # @return [void]
     def each_filtered
-      if ignore_private
-        if ignore_public
-          yield Metasploit::Framework::Credential.new(public: nil, private: nil, realm: realm)
-        else
-          yield Metasploit::Framework::Credential.new(public: username, private: nil, realm: realm)
-        end
-      elsif password_spray
-        each_unfiltered_password_first do |credential|
-          next unless self.filter.nil? || self.filter.call(credential)
-
-          yield credential
-        end
-      else
-        each_unfiltered_username_first do |credential|
-          next unless self.filter.nil? || self.filter.call(credential)
+      each_unfiltered do |credential|
+        next unless self.filter.nil? || self.filter.call(credential)
 
-          yield credential
-        end
+        yield credential
       end
     end
 
     alias each each_filtered
 
+    def each_unfiltered(&block)
+      prepended_creds.each { |c| yield c }
+
+      if anonymous_login
+        yield Metasploit::Framework::Credential.new(public: '', private: '', realm: realm, private_type: :password)
+      end
+
```

Comment:
I believe this change is required to align with some recent changes made as part of this PR: https://github.com/rapid7/metasploit-framework/pull/19843

Specifically this code:
```suggestion

      if ignore_private
        if ignore_public
          yield Metasploit::Framework::Credential.new(public: nil, private: nil, realm: realm)
        else
          yield Metasploit::Framework::Credential.new(public: username, private: nil, realm: realm)
        end
      end
      
```